lib/signs_ui_web/controllers/messages_controller.ex

```diff
@@ -87,4 +87,97 @@ defmodule SignsUiWeb.MessagesController do
   def create(conn, _params) do
     send_resp(conn, 201, "Ignoring unknown message.")
   end
+
+  def background(conn, _params) do
+    with {:ok, visual_zones} <- parse_zones(conn, "visual_zones"),
+         {:ok, visual_data} <- parse_visual_data(conn),
+         {:ok, expiration} <- parse_expiration(conn) do
+      scu_id =
+        Enum.find_value(conn.req_headers, fn {key, value} ->
+          if(key == "x-scu-id", do: value)
+        end)
+
+      zone = MapSet.to_list(visual_zones) |> hd()
```

Comment:
**quibble (non-blocking):** `Enum.at/2` would allow you to fetch a single value without iterating over the entire MapSet to convert it to a list first. Not sure how much that matters though.

```suggestion
      zone = Enum.at(visual_zones, 0)
```